pkgs/applications/science/math/mathics/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib, fetchFromGitHub, fetchPypi, buildPythonPackage, buildPythonApplication, isPy3k
+, glibcLocales, mpmath
+, sympy, numpy, pint, django_3, dateutil, llvmlite, requests, palettable, setuptools, wordcloud, scikitimage, python
+, substituteAll
+}:
+
+let
+  deps = [
+    sympy numpy pint django_3 mpmath dateutil llvmlite requests palettable setuptools wordcloud scikitimage
+  ];
+  pythonEnv = python.withPackages (p: deps);
+in
+  buildPythonApplication rec {
+    pname = "mathics";
+    version = "1.1.1";
+
+    src = fetchFromGitHub {
+      owner = "mathics";
+      repo = "Mathics";
+      rev = version;
+      sha256 = "sha256-6VCz/pd2kRUkezCoXjqvPYhbzHojHQZ7y//NAFrPHas=";
+    };
+
+    propagatedBuildInputs = deps;
+
+    postPatch = ''
+      substituteInPlace mathics/server.py \
+        --replace "sys.executable, " ""
+    '';
+
+    # We run tests at the `postInstall` stage. This also installs documentation.
+    doCheck = false;
+
+    postInstall = ''
+      rm $out/bin/dmathics*
+      SANDBOX=true python mathics/test.py --keep-going
```

Comment:
Please move this to test phase.

Reply:
The main purpose of this line is not to run tests, but to generate documentation.

Reply:
Totally visible by the name. Maybe add a comment so that the next one does not suggest the same.